backend/src/packages/chaiNNer_pytorch/pytorch/processing/upscale_image.py

```diff
@@ -39,13 +40,28 @@ def upscale(
         device = torch.device(options.full_device)
 
         def estimate():
+            element_size = 2 if use_fp16 else 4
+            model_bytes = sum(p.numel() * element_size for p in model.parameters())
+
             if "cuda" in options.full_device:
                 mem_info: Tuple[int, int] = torch.cuda.mem_get_info(device)  # type: ignore
                 free, _total = mem_info
-                element_size = 2 if use_fp16 else 4
-                model_bytes = sum(p.numel() * element_size for p in model.parameters())
                 budget = int(free * 0.8)
 
+                return MaxTileSize(
+                    estimate_tile_size(
+                        budget,
+                        model_bytes,
+                        img,
+                        element_size,
+                    )
+                )
+
+            if is_arm_mac:
+                total_memory = psutil.virtual_memory().total
+                reserved_system_memory = options.reserved_system_memory * (1024**2)
+                budget = int(total_memory - reserved_system_memory)
```

Comment:
I would suggest lowering the budget to 80% just like we do for CUDA. 

`estimate_tile_size` really does give back an estimate. The estimated tile size might use a lot more RAM than we allowed. This isn't a huge issue with CUDA because we just have to restart upscaling when we run out of VRAM, but it's a big issue when the same thing happens with RAM. 

Look at it like this: when we estimate a tile size smaller than optimal, upscaling just takes a bit longer, but when we estimate a tile size that is too large, the whole system will stutter (or whatever happens when a mac runs out of memory).

Reply:
The budget is determined by the value set in `reserved_system_memory`.

On my system 32 GB, it could be 20% when it set it to the max possible value (25.6 GB) and 75% when set to the min possible value (8 GB).

But you're right, could be more than 80% on systems with 64 GB and higher.

Reply:
Fixed, the budget can now be between 20% and 80% of the system memory.

Reply:
I also added a fail-safe. So the budget will always be between 20% and 80% of the total system memory.